src/averageAges.js

```diff
@@ -37,7 +51,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people
+    .filter((person) => withChildren
+      ? person.sex === 'f' && people.find(
+        (child) => person.name === child.mother
+      )
+      : person.sex === 'f');
+
+  return women
+    .map((peopl) => peopl.died - peopl.born)
+    .reduce((a, b) => a + b) / women.length;
```

Comment:
you are doing the same (map and reduce) twice - move it to a separate function and reuse